models/experimental/hugging_face.py

```diff
@@ -122,33 +122,74 @@ class HFTransformerCore(torch.nn.Module):
 
     def __init__(self, model_cfg):
         super().__init__()
+        
         flash_attn = model_cfg.get("flash_attention", False)
         if flash_attn:
             attn_impl = "flash_attention_2"
         else:
             attn_impl = "eager"
+
+        ## set the model name (has to be from huggingface)
+        model_name = model_cfg["model_string"]
+
+        ## load the model from the model hub
         self.model = AutoModelForCausalLM.from_pretrained(
-            model_cfg["model_string"],
+            model_name,
             trust_remote_code=True,
             attn_implementation=attn_impl,
             torch_dtype=torch.float16,
-        )
+            )
+
+        ## Store the model's lm_head separately and set it to None
+        if hasattr(self.model, 'lm_head'):
+            self.lm_head = self.model.lm_head
+            self.model.lm_head = None
```

Comment:
please test this with a number of models... i'm not sure if it is part of the huggingface api and thus sth that we can assume across models